### PR TITLE
34186 Serialize demographics confirmation data

### PR DESCRIPTION
**Pull request by kanchanasuriya**
## Description of change

Implementation for returning patient demographic confirmation status to front end for facilitating day-of-check-in app to disable demographic pages to the user

## Original issue(s)
department-of-veterans-affairs/va.gov-team#34186

## Things to know about this PR

- [x] rspec
- [x] rubocop

### CODE REVIEW

File: modules/check_in/app/serializers/check_in/v2/appointment_data_serializer.rb [hunk 1]
```diff
@@ -46,7 +46,22 @@ class AppointmentDataSerializer
...
+            demographicsNeedsUpdate: raw_confirmation[:demographicsNeedsUpdate],
+            demographicsConfirmedAt: raw_confirmation[:demographicsConfirmedAt],
+            nextOfKinNeedsUpdate: raw_confirmation[:nextOfKinNeedsUpdate],
+            nextOfKinConfirmedAt: raw_confirmation[:nextOfKinConfirmedAt],
+            emergencyContactNeedsUpdate: raw_confirmation[:emergencyContactNeedsUpdate],
+            emergencyContactConfirmedAt: raw_confirmation[:emergencyContactConfirmedAt]
+          }
+          { demographics: demographics,
+            appointments: appointments,
+            patientDemographicsStatus: demographics_confirmation }
```
Comment (randomsync) [inline modules/check_in/app/serializers/check_in/v2/appointment_data_serializer.rb:61]: minor (just a styling suggestion):
```suggestion
          {
            demographics: demographics,
            appointments: appointments,
            patientDemographicsStatus: demographics_confirmation
          }
```

Comment (kanchanasuriya) [reply modules/check_in/app/serializers/check_in/v2/appointment_data_serializer.rb]: Sure Gaurav, addressed in [98323](https://github.com/department-of-veterans-affairs/vets-api/pull/8722/commits/983233865741a6bb9066c616d695d167e4b609e8)

**Review by randomsync — APPROVED** [review]

**Review by randomsync — COMMENTED** [review]
Actually, doesn't `check_in_experience_demographics_confirmation_enabled` need to be added to features.yml or does it already exist?

**Comment by kanchanasuriya** [comment]
> Actually, doesn't `check_in_experience_demographics_confirmation_enabled` need to be added to features.yml or does it already exist?

Thank you, thats a good catch. I think it got removed in one of the rebase before pushing.  Added the feature toggle back as part of this [commit](https://github.com/department-of-veterans-affairs/vets-api/pull/8722/commits/38d00c35c2d5964aa738c975d61806da9241f7d7).

**Review by randomsync — APPROVED** [review]
lgtm

**Review by thilton-oddball — APPROVED** [review]

**Pull request merged**